packages/cli/src/index.ts

```diff
@@ -99,12 +99,38 @@ export async function build(_options: CliOptions) {
 
   await generate(options)
 
-  startWatcher()
+  await startWatcher()
+
+  function transformFiles(sources: { id: string; code: string; transformedCode?: string | undefined }[], enforce: SourceCodeTransformerEnforce = 'default') {
+    return Promise.all(
+      sources.map(({ id, code, transformedCode }) => new Promise<{ id: string; code: string; transformedCode: string | undefined }>((resolve) => {
+        applyTransformers(ctx, code, id, enforce)
+          .then((transformsRes) => {
+            resolve({ id, code, transformedCode: transformsRes?.code || transformedCode })
+          })
+      })))
+  }
 
   async function generate(options: ResolvedCliOptions) {
+    const sourceCache = Array.from(fileCache).map(([id, code]) => ({ id, code }))
+
     const outFile = resolve(options.cwd || process.cwd(), options.outFile ?? 'uno.css')
-    const { css, matched } = await uno.generate(
-      [...fileCache].join('\n'),
+
+    const preTransform = await transformFiles(sourceCache, 'pre')
+    const defaultTransform = await transformFiles(preTransform)
+    const postTransform = await transformFiles(defaultTransform, 'post')
```

Comment:
I personally think it's unnecessary, `enfore` is more used as a plugin property, like `vite`

Reply:
But the `transformerVariantGroup`, `transformerAttributifyJsx`, `transformerCompileClass` default enforce is `pre`, others transformers default enforce is `undefined`, `applyTransformers` will filter the corresponding transformer

Reply:
You are right. 😅